admin-dev/themes/new-theme/js/pages/product/carrier/CarrierSelector.vue

```diff
@@ -0,0 +1,164 @@
+          @click="clearAllSelected"
+        >
+          <i class="material-icons">close</i>
+          {{ $t('allCarriers.label') }}
+        </button>
+      </div>
+    </div>
+    <div
+      v-if="selectedCarrierIds.length"
+      id="selected-carriers"
+    >
+      <span>
+        <ul>
+          <li
+            v-for="(selectedCarrier, key) in selectedCarriers"
+            :key="key"
+          >
+            {{ selectedCarrier.label }}<span v-if="key !== selectedCarriers.length -1 ">, </span>
+          </li>
+        </ul>
+      </span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  import CheckboxesDropdown from '@app/components/checkboxes-dropdown/CheckboxesDropdown.vue';
+  import {defineComponent, PropType} from 'vue';
+  import {Choice} from '@app/components/checkboxes-dropdown/types';
+  import EventEmitter from '@components/event-emitter';
+  import ProductEventMap from '@pages/product/product-event-map';
+  import ProductMap from '@pages/product/product-map';
+
+  export default defineComponent({
+    name: 'CarrierSelector',
+    data(): {
+      selectedCarrierIds: number[],
+      modifyAllShopsVisible: boolean,
+    } {
+      return {
+        selectedCarrierIds: [],
+        modifyAllShopsVisible: false,
+      };
+    },
+    props: {
+      initialCarrierIds: {
+        type: Array as PropType<number[]>,
+        required: true,
+      },
+      carrierChoices: {
+        type: Array as PropType<Choice[]>,
+        required: true,
+      },
+      choiceInputName: {
+        type: String,
+        required: true,
+      },
+      eventEmitter: {
+        type: Object as PropType<typeof EventEmitter>,
+        required: true,
+      },
+    },
+    components: {
+      CheckboxesDropdown,
+    },
+    mounted() {
+      this.selectedCarrierIds = this.initialCarrierIds;
+    },
+    computed: {
+      selectedCarriers(): Choice[] {
+        return this.carrierChoices.filter((carrier: Choice) => this.selectedCarrierIds.includes(carrier.id));
+      },
+      getLabel(): string {
+        return this.selectedCarrierIds.length > 0
+          ? this.$t('selectedCarriers.label')
+          : this.$t('allCarriers.label');
+      },
+    },
+    methods: {
+      addCarrier(carrier: Choice): void {
+        this.selectedCarrierIds.push(carrier.id);
+      },
+      removeCarrier(carrier: Choice): void {
+        this.selectedCarrierIds = this.selectedCarrierIds.filter(
+          (id: number) => carrier.id !== id,
+        );
+      },
+      clearAllSelected(): void {
+        this.selectedCarrierIds = [];
+        this.eventEmitter.emit(ProductEventMap.shipping.clearAllCarriers);
+        const carrierSelector = <HTMLDivElement> document.querySelector(ProductMap.shipping.carrierSelectorContainer);
+        // this event will allow modify_all_shops checkbox to stay visible after clearing all carriers
+        carrierSelector.dispatchEvent(new Event('change'));
```

Comment:
I think this change event should be triggered by the select input, not the container div
And if this works maybe the `clearAllCarriers` will then become useless as we discussed